Week 04/Lecture 08/Assignment 02/assignment2/src/main/java/example/lecture8/assignment2/repository/EmployeeRepository.java

```diff
@@ -0,0 +1,53 @@
+package example.lecture8.assignment2.repository;
+
+import example.lecture8.assignment2.model.Employee;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+@Repository
+public class EmployeeRepository {
+    
+    private final JdbcTemplate jdbcTemplate;
+
+    public EmployeeRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private static final class EmployeeRowMapper implements RowMapper<Employee> {
+        @Override
+        public Employee mapRow(ResultSet rs, int rowNum) throws SQLException {
+            Employee emp = new Employee();
+            emp.setId(rs.getInt("id"));
+            emp.setName(rs.getString("name"));
+            emp.setDepartment(rs.getString("department"));
+            emp.setAge(rs.getInt("age"));
+            return emp;
+        }
+    }
+
+    public List<Employee> findAll() {
+        return jdbcTemplate.query("SELECT * FROM employee", new EmployeeRowMapper());
+    }
+
+    public Employee findEmployeeById(int id) {
+        return jdbcTemplate.queryForObject("SELECT * FROM employee WHERE id = ?", new EmployeeRowMapper(), id);
```

Comment:
It should be return exception named EmptyResultDataAccessException, it handle with try and catch if there any employee with provided id. It will return string that contain message "Employee with id {id} doesn't exist". Done revised the code.